experimental/telemetry/tracker.go

```diff
@@ -105,3 +108,28 @@ func (t *tracker) TrackUserEvent(event, userID string, properties map[string]int
 	properties["UserActualID"] = userID
 	return t.TrackEvent(event, properties)
 }
+
+type telemetryAPIRequest struct {
+	Event      string
+	Properties map[string]interface{}
+}
+
+func (t *tracker) HandleClientEvent(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	var telemetryRequest *telemetryAPIRequest
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&telemetryRequest)
+	if err != nil {
+		http.Error(w, "Unable to decode JSON", http.StatusBadRequest)
```

Comment:
It's a bummer that there is no entity to log the error

Reply:
What we can do is to return the error, and leave the http handling to the caller. What do you think?

Reply:
Leaving it to the caller would push the error to the webapp in most of the cases. For debugging it's more useful to log the error in the server logs.

Reply:
Right now the tracker does not have "logging functions", and I would prefer to leave it like that. I made some changes to return the error.